tests/suites/test_suite_psa_crypto.function

```diff
@@ -8447,17 +8464,19 @@ exit:
 
 /* BEGIN_CASE */
 void derive_input(int alg_arg,
-                  int step_arg1, int key_type_arg1, data_t *input1,
-                  int expected_status_arg1,
-                  int step_arg2, int key_type_arg2, data_t *input2,
-                  int expected_status_arg2,
-                  int step_arg3, int key_type_arg3, data_t *input3,
-                  int expected_status_arg3,
+                  int step_arg1, int key_type_arg1, int input_type_arg1,
```

Comment:
This function already takes an annoyingly large number of arguments. Rather than make it even worse, I'd prefer to combine the information integer/not-integer with the existing parameter that indicates type-of-key/not-a-key. All it takes is to use a value for `key_type_argN` that is neither a valid key type, nor an invalid key type that we might use for negative testing, nor `PSA_KEY_TYPE_NONE` which we use to convey not-a-key-but-bytes.

So something like
```
#define INPUT_INTEGER 0x10000 // out of range of psa_key_type_t
```
in the function file, and use that for the `key_type_argN` parameters to say that the `inputN` parameter must be parsed as an integer and the test function must call `psa_key_derivation_input_integer`.